src/utils/validations/component/validations/form-with-validations.hoc.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+const ValidationsContext = React.createContext();
+
+const withValidations = (WrappedComponent) => {
+  return class extends React.Component {
+    state = {
+      errorCount: 0,
+      warningCount: 0,
+      infoCount: 0
+    }
+
+    inputs = {};
+
+    validationTypes = ['error', 'warning', 'info'];
+
+    addInput = (name, validate) => {
+      this.inputs[name] = validate;
+    }
+
+    removeInput = (name) => {
+      delete this.inputs[name];
+    }
+
+    adjustCount = (type, validationResult) => {
```

Comment:
I think that it could be a boolean value (maybe `hasError`?), as `validationResult` is an ErrorStatus passed from the validation.

Reply:
Good point seems redundant to pass the actual error up here, I'll just pass in a boolean. I'm going to rename it `hasFailed` as it covers all of the validation subsets that may have failed